src/PluginClientFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Http\Client\Common;
+
+use Http\Client\HttpAsyncClient;
+use Http\Client\HttpClient;
+
+/**
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ */
+final class PluginClientFactory
+{
+    /**
+     * @var callable
+     */
+    private static $factory;
```

Comment:
As the stuff is marked internal, we choosed to avoid over engineering and to not use an interface.